### DISCUSSION TITLE
How to index an entire 1d array from a 2d array?

### DISCUSSION THREAD

**Discussion by TrevorSatori** [rust-ndarray/ndarray#1214] [Q&A] [answered]
Sorry if this is noobie, but I have been stuck on this for hours. In the docs it makes very clear how to index a specific value from a 2d array arr[[0,1]] = 5, but how would I replace a 1d array (row) inside a 2d array?

I believe it should look something like this.

let mut x: Array2<f64> = Array::zeros((10 , 5));

x[0] += [0,0,0,0,0];

please help.

**Comment by jturner314** [comment, marked as answer]
I'd suggest reading the [Slicing](https://docs.rs/ndarray/0.15.6/ndarray/struct.ArrayBase.html#slicing) and [Subviews](https://docs.rs/ndarray/0.15.6/ndarray/struct.ArrayBase.html#subviews) docs. The [`ndarray` for NumPy users](https://docs.rs/ndarray/0.15.6/ndarray/doc/ndarray_for_numpy_users/index.html) guide may also be of interest.

Here's some sample code for your specific question:

```rust
use ndarray::prelude::*;
use std::ops::AddAssign;

fn main() {
    let mut x: Array2<i32> = Array::zeros((10, 5));
    
    // The examples below use `.row_mut()`. Other options include
    // `.slice_mut()` and `.index_axis_mut()`.
    
    // Assign a row.
    x.row_mut(0).assign(&aview1(&[0,0,0,0,0]));
    
    // Fill all elements of a row with the same value.
    x.row_mut(0).fill(0);
    
    // Add to a row in-place (+= operation). Note the `use std::ops::AddAssign` above.
    x.row_mut(0).add_assign(&aview1(&[0,0,0,0,0]));
}
```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=928563a06e3b21545dc7bd4307977e15))

**Reply by TrevorSatori** [reply]
Thanks so much for this.
